### DISCUSSION TITLE
unusual disk activity at startup

### DISCUSSION THREAD

**Discussion by xgiovio** [openzfs/zfs#13069] [Q&A]
Hello, I have centos 7, latest kernel and latest zfs. All works as it should. I have 3 pools, each of them with 600 snapshots.

What i noticed in the past months is that everytime I boot the os (single ssd), all pools are mounted but using "iostat -y 5" I can see activity of about 1 megabyte/s on each drive for about 5/10 minutes until all of them go to 0 mb/s. I discovered it because i have the system near me af after the start up i can feel the noise of all the discs together working.

I don't undersand what is going on. Is it caching from disks to memory all the snapshots references? What cauld be the problem? Considering that i start and shutdown the system everyday , I would't put too much stress on the disks.

Thank you in advance for your help.

**Comment by xgiovio** [comment]
i did, used iostat and zpool iostat to monitor the writing and reading. Iotop to monitor the processes.

The processes are:

systemd udev

zed -f

about 1mb/s

**Comment by IvanVolosyuk** [comment]
Do you use l2arc by any chance? I think it can asynchronously load data
from persistent l2arc.

**Comment by xgiovio** [comment]
Yes I use the l2arc. If the l2arc is persistent why is it loading data each time?

**Comment by IvanVolosyuk** [comment]
It should load the mapping from block pointers to locations in l2arc. It
doesn't load the data itself, just l2arc metadata.

**Comment by xgiovio** [comment]
Is it possible to modify this behaviour? I would like to disable this 
load from the disks at each startup. Can't the metadata be stored 
elsewhere or remain on l2arc?

Giovanni Di Grezia - VFX Artist and Software Developer

------ Original Message ------
From: "Ivan Volosyuk" ***@***.***>
To: "openzfs/zfs" ***@***.***>
Cc: "Giovanni Di Grezia" ***@***.***>; "Author" 
***@***.***>
Sent: 06/02/2022 10:44:11
Subject: Re: [openzfs/zfs] unusual disk activity at startup (Discussion 
#13069)

>
>It should load the mapping from block pointers to locations in l2arc. 
>It
>doesn't load the data itself, just l2arc metadata.
>
>On Sun, Feb 6, 2022 at 7:36 PM Giovanni Di Grezia ***@***.***>
>wrote:
>
> > Yes I use the l2arc. If the l2arc is persistent why is it loading 
>data
> > each time?
> >
> > > On 6 Feb 2022, at 08:32, Ivan Volosyuk ***@***.***> wrote:
> > >
> > > ﻿
> > > Do you use l2arc by any chance? I think it can asynchronously load 
>data
> > > from persistent l2arc.
> > >
> > > On Sun, Feb 6, 2022 at 9:57 AM Giovanni Di Grezia ***@***.***>
> > > wrote:
> > >
> > > > i did, used iostat and zpool iostat to monitor the writing and 
>reading.
> > > > Iotop to monitor the processes.
> > > >
> > > > The processes are:
> > > >
> > > > systemd udev
> > > >
> > > > zed -f
> > > >
> > > > about 1mb/s
> > > >
> > > > —
> > > > Reply to this email directly, view it on GitHub
> > > > <
> > 
>https://github.com/openzfs/zfs/discussions/13069#discussioncomment-2118152
> > >,
> > > > or unsubscribe
> > > > <
> > 
>https://github.com/notifications/unsubscribe-auth/ABXQ6HO6SIV4CBZZVJ2KJPTUZWTNHANCNFSM5NUP32EQ
> > >
> > > > .
> > > > Triage notifications on the go with GitHub Mobile for iOS
> > > > <
> > 
>https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> > >
> > > > or Android
> > > > <
> > 
>https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
> > >.
> > > >
> > > > You are receiving this because you are subscribed to this
> > thread.Message
> > > > ID: ***@***.***>
> > > >
> > > —
> > > Reply to this email directly, view it on GitHub, or unsubscribe.
> > > Triage notifications on the go with GitHub Mobile for iOS or 
>Android.
> > > You are receiving this because you authored the thread.
> >
> > —
> > Reply to this email directly, view it on GitHub
> > 
><https://github.com/openzfs/zfs/discussions/13069#discussioncomment-2119115>,
> > or unsubscribe
> > 
><https://github.com/notifications/unsubscribe-auth/ABXQ6HLGAUEBDPXL7KOISFDUZYXINANCNFSM5NUP32EQ>
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > 
><https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> > or Android
> > 
><https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
> >
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
>—
>Reply to this email directly, view it on GitHub 
><https://github.com/openzfs/zfs/discussions/13069#discussioncomment-2119288>, 
>or unsubscribe 
><https://github.com/notifications/unsubscribe-auth/AAXRGXTYQEMHQ26KDSUI7NLUZY7GXANCNFSM5NUP32EQ>.
>Triage notifications on the go with GitHub Mobile for iOS 
><https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
>or Android 
><https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>You are receiving this because you authored the thread.Message ID: 
***@***.***>
>

**Comment by IvanVolosyuk** [comment]
You can disable persistent l2arc using zfs.l2arc_rebuild_enabled=0. You
will start from a clean l2arc without any loading from it. When the system
boots it has to find out what is stored in persistent l2arc, thus this i/o
happens. But you can of course disable it.

**Reply by xgiovio** [reply]
even if the settings off the behaviour is the same.
I set it in /etc/modprobe.d/zfs.conf and after reboot i can see that /sys/module/zfs/parameters/l2arc_rebuild_enabled is 0
however the reading from disks at startups using zpool iostat -v 1 remains. I need to understamd what is the problem.

**Comment by xgiovio** [comment]
this is definitely a bug
solved removing the autoexpand property and setting it to off

**Comment by xgiovio** [comment]
The issue i already present #7132
